inspirehep/modules/forms/static/js/forms/books_typeahead.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of INSPIRE.
+ * Copyright (C) 2017 CERN.
+ *
+ * INSPIRE is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * INSPIRE is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this licence, CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+define([
+  'js/forms/extended_typeahead'
+], function(ExtendedTypeahead) {
+
+  function booksTypeahead($element) {
+    $element.extendedTypeahead({
+      suggestionTemplate: Hogan.compile(
+        '<b>{{ title }}</b><br>' +
+        '<small>' +
+        '{{ author_list }}<br>' +
+        '{{ date }}<br>'+
+        '</small>'
+      ),
+      selectedValueTemplate: Hogan.compile(
+        'Title: {{ title }}, Authors: {{author_list }}, Date: {{date}}'
+      ),
+      cannotFindMessage: 'Cannot find this book in our database.',
+      extractRawValue: function(data) {
+        return data.control_number;
+      },
+      displayKey: null,
+      displayfn: function(obj) {
+        return obj;
+      },
+      dataEngine: new Bloodhound({
+        name: 'books',
+        remote: {
+          url: '/api/literature?q=bookautocomplete:%QUERY*',
+          filter: function(response) {
+            return $.map(response.hits.hits, function(el) {
+              el.metadata.title = el.metadata.titles[0].title;
+              if(typeof el.metadata.isbns !== 'undefined'){
+                el.metadata.isbn = el.metadata.isbns[0].value;
+              }
+              if(typeof el.metadata.earliest_date !== 'undefined'){
+                el.metadata.date = el.metadata.earliest_date;
+              }(function(a) {return a.foo;});
+              if(typeof el.metadata.authors !== 'undefined'){
+                  el.metadata.author_list = el.metadata.authors.map( function(a) {return a.full_name.replace(',','')});
```

Comment:
Well since im separating the authors with commas im removing the name surname comma separator.